### PR TITLE
bump dotnet and ef to 1.1.0-preview1

### PR DESCRIPTION
**Pull request by caleblloyd**
- Upgrade to [.NET Core 1.1 Preview 1](https://blogs.msdn.microsoft.com/dotnet/2016/10/25/announcing-net-core-1-1-preview-1/)
- Upgrade to [Entity Framework Core 1.1 Preview 1](https://blogs.msdn.microsoft.com/dotnet/2016/10/25/announcing-entity-framework-core-1-1-preview-1/)
- Visual Studio and Command Line Tooling still depends on .NET Core 1.0.0 until https://github.com/aspnet/EntityFramework/issues/6897 is fixed

### CODE REVIEW

**Comment by caleblloyd** [comment]
I've modified the CI a bit:

- AppVeyor now builds 4.5.1 to check for build errors, not Travis CI, so we don't have to go through the long process of installing mono on Travis CI
- KoreBuild on AppVeyor is used to install .NET Core, but tests are manually selected giving us better control of how to run our tests.  I had to do this to get the configuration switches working on tests (`dotnet test -c Release`)

Our coverage remains the same as it was before, it is just slightly different.

File: build-run.ps1 [hunk 1]
```diff
@@ -33,18 +33,13 @@ cd $PSScriptRoot
 $repoFolder = $PSScriptRoot
 $env:REPO_FOLDER = $repoFolder
 
-if ($env:BuildRunner -eq "MyGet"){
-    Remove-Item -Force -Recurse (Join-Path $PSScriptRoot (Join-Path "test" "Pomelo.EntityFrameworkCore.MySql.PerfTests"))
-}
-
-$koreBuildZip="https://github.com/PomeloFoundation/KoreBuild/releases/download/1.0.0/KoreBuild-1.0.0.zip"
+$koreBuildZip="https://github.com/aspnet/KoreBuild/archive/dev.zip"
```
Comment (Kagamine) [inline build-run.ps1]: Are you sure the aspnet/KoreBuild/archive/dev.zip will clone the 1.1.0-preview1-final environment(not 1.1.0-*)?

**Comment by Kagamine** [comment]
There are many differences between mono and .NET, so I want to keep mono.

File: build-run.ps1 (hunk 1 above)
Comment (caleblloyd) [inline build-run.ps1]: Right now we are just using the install part of KoreBuild, which installs the latest preview environment, `dotnet-dev-win-x64.1.0.0-preview2.1-003155.zip`.  This is the same version as listed on the [Announcing .NET Core 1.1 Preview 1](https://blogs.msdn.microsoft.com/dotnet/2016/10/25/announcing-net-core-1-1-preview-1/) page.

**Comment by caleblloyd** [comment]
I'll add back a mono build through Docker

**Comment by caleblloyd** [comment]
Added mono back to Travis CI.  Look good to merge?

**Comment by Kagamine** [comment]
@caleblloyd There were something error in MyGet build service after merging this.

**Comment by caleblloyd** [comment]
@Kagamine fixed MyGet and published [1.1.0-preview1-10001](https://www.nuget.org/packages/Pomelo.EntityFrameworkCore.MySql/1.1.0-preview1-10001)

**Pull request merged**